cmd/operator-sdk/bundle/validate.go

```diff
@@ -99,66 +120,98 @@ func (c bundleValidateCmd) validate(args []string) error {
 	if len(args) != 1 {
 		return errors.New("an image tag or directory is a required argument")
 	}
+	if c.outputFormat != output.JSONAlpha1 && c.outputFormat != output.Text {
+		return fmt.Errorf("invalid value for output flag: %v", c.outputFormat)
+	}
 	return nil
 }
 
 func (c *bundleValidateCmd) addToFlagSet(fs *pflag.FlagSet) {
 	fs.StringVarP(&c.imageBuilder, "image-builder", "b", "docker",
 		"Tool to extract container images. One of: [docker, podman]")
+	fs.StringVarP(&c.outputFormat, "output", "o", output.Text,
+		"Result format for results. One of: [text, json-alpha1]")
+
+	// It is hidden because it is an alpha option
+	// The idea is the next versions of Operator Registry will return a List of errors
+	if err := fs.MarkHidden("output"); err != nil {
+		panic(err)
+	}
 }
 
-func (c bundleValidateCmd) run() (err error) {
+func (c bundleValidateCmd) run(logger *log.Entry) (res output.Result, err error) {
 	// Set directory, either supplied directly or a temp dir used to unpack
 	// the image.
 	dir := c.directory
 	if c.imageTag != "" {
 		dir, err = ioutil.TempDir("", "bundle-")
 		if err != nil {
-			return err
+			return res, err
 		}
 		defer func() {
 			if err = os.RemoveAll(dir); err != nil {
-				log.Errorf("Error removing temp bundle dir: %v", err)
+				logger.Errorf("Error removing temp bundle dir: %v", err)
 			}
 		}()
 	}
 	if dir, err = filepath.Abs(dir); err != nil {
-		return err
+		return res, err
 	}
 
 	// Set up logger.
 	fields := log.Fields{"bundle-dir": dir}
 	if c.imageTag != "" {
 		fields["container-tool"] = c.imageBuilder
 	}
-	logger := log.WithFields(fields)
-	if viper.GetBool(flags.VerboseOpt) {
-		log.SetLevel(log.DebugLevel)
-	}
-
+	logger = log.WithFields(fields)
 	val := bundle.NewImageValidator(c.imageBuilder, logger)
 
 	// Pull image if a tag was passed.
 	if c.imageTag != "" {
 		logger.Info("Unpacked image layers")
 		err = val.PullBundleImage(c.imageTag, dir)
 		if err != nil {
-			logger.Fatalf("Error to unpacking image: %v", err)
+			return res, fmt.Errorf("error unpacking image: %w", err)
 		}
 	}
 
+	// Create a new res containing validation errors and info msgs, if any
+	res = output.Result{}
+	res.Passed = true // Init without errors
+
 	// Validate bundle format.
 	if err = val.ValidateBundleFormat(dir); err != nil {
-		logger.Fatalf("Bundle format validation failed: %v", err)
+		addBundleValErrorToOutput(err, &res)
```

Comment:
This seems more convoluted than collecting errors and evaluating them in a constructor like `NewResult(errs...)`, which you then return at the end of the function.

Reply:
Evaluating specifically Bundle Validation errors is just part of the bundle context and then, IMO is not could not be in the NewResult. I mean, it is very specific of this implementation when the Result is generic and can be used across the project.